File: PetCare.Server/DTOs/VaccineDTO.cs
using AutoMapper;
using PetCare.Shared.Entities;

namespace PetCare.Server.DTOs
{
    public class VaccineDTO
    {
        public Guid Id { get; set; }
        public Guid PetId { get; set; }
        public required string Name { get; set; }
        public DateTime AdministrationDate { get; set; }
        public DateTime NextDueDate { get; set; }
        public string? Notes { get; set; }
    }

    public class VaccineProfile : Profile
    {
        public VaccineProfile()
        {
            CreateMap<Vaccine, VaccineDTO>().ReverseMap();
        }
    }
}

File: petcare-web-2/src/types/dtos/vaccineDTO.ts
export interface VaccineDTO {
  id: string;
  petId: string;
  name: string;
  administrationDate: Date;
  notes?: string;
}

export interface UpcomingVaccineDTO {
  id: string;
  petId: string;
  name: string;
  date: Date;
  notes?: string;
}
